### DISCUSSION TITLE
Is it possible to hard set password from a cron job?

### DISCUSSION THREAD

**Discussion by JunoNgx** [pocketbase/pocketbase#4738] [Q&A] [answered]
Context: I have a live demo account (very similar to PocketBase website, in fact), whose data I'd like to reset every couple of hours.

I have managed to implemented this with `cronAdd` (deleting related records and recreating them again), though as an extra piece of assurance, I'd like to reset the password as well.

It is possible to just hard set the password programmatically?

I've tried both with and without validation:

```
const form = new RecordUpsertForm($app, demoUser);
form.loadData({
    displayName: "Jay Doe",
    password: "password123", // throws error, will ask for `confirmPassword` and `oldPassword`
});
form.submit();
```

```
// Does not do anything
demoUser.set("displayName", "Jay Doe");
demoUser.set("password", "password123")
```

Is there any way I can do this, or this is something I'd have to live with?

Edit: I forgot to call `$app.dao().saveRecord(demoUser);` in the method without validation, but looks like this has no effect on the password, either.

**Comment by ganigeorgiev** [comment, marked as answer]
If you want to use the form, you can try with:

```js
const form = new RecordUpsertForm($app, demoUser);
form.setFullManageAccess(true) // allows changes to system fields like email, verified, password, etc. without the user confirmation
form.loadData({
    displayName: "Jay Doe",
    password: "password123",
    passwordConfirm: "password123",
});
form.submit();
```

Or if you want to change the record model directly you have to use the `setPassword("password123")` setter:

```js
// this behind the scenes will auto update the passwordHash and tokenKey fields.
demoUser.setPassword("password123")
$app.dao().saveRecord(demoUser);
```

This is already normalized as part of the [refactoring](https://github.com/pocketbase/pocketbase/discussions/4355) and `record.Set("password", "123456")` will automatically call under the hood the same operations as `record.SetPassword("123456")`.
